### PR TITLE
STRM-7: Added modules & media type fields

### PR DESCRIPTION
**Pull request by mvpsatyam**

### CODE REVIEW

File: config/taxonomy.vocabulary.media.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+langcode: en
+status: true
+dependencies: {  }
+name: Media
+vid: media
```
Comment (malabya) [inline config/taxonomy.vocabulary.media.yml]: Label -  `Tags - Media`
vid - tags_media

File: recipe.yml [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
+recipes:
+  - core/recipes/document_media_type
+  - core/recipes/remote_video_media_type
+  - core/recipes/local_video_media_type
 install:
   - media
   - media_library
   - filefield_paths
   - focal_point
   - media_entity_download
```
Comment (malabya) [inline recipe.yml:12]: Give permissions to anonymous & authenticated to download media

File: composer.json [hunk 3]
```diff
@@ -12,6 +12,7 @@
         "drupal/focal_point": "^2.1",
         "drupal/webp": "^1.0@RC",
         "drupal/media_entity_download": "^2.3",
-        "drupal/filefield_paths": "^1.0@beta"
+        "drupal/filefield_paths": "^1.0@beta",
+        "drupal/lazy": "3.x-dev"
     }
```
Comment (malabya) [inline composer.json:19]: add https://www.drupal.org/project/media_file_delete and configure it to delete the file

File: config/core.entity_form_display.media.image.default.yml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    - field.field.media.image.field_media_tags
+    - image.style.thumbnail
+    - media.type.image
+  module:
+    - image
+    - path
+    - select2
+    - text
+_core:
+  default_config_hash: w4xWcmH5bjJmqILkXyy8v3ql5SNq9_jUE7MalSOyi18
```
Comment (malabya) [inline config/core.entity_form_display.media.image.default.yml]: Remove the config hash

File: config/core.entity_form_display.media.image.media_library.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    - field.field.media.image.field_media_caption
+    - field.field.media.image.field_media_credit
+    - field.field.media.image.field_media_image
+    - field.field.media.image.field_media_tags
+    - image.style.thumbnail
+    - media.type.image
+  module:
+    - image
+_core:
+  default_config_hash: iejfMkPx4bW_bOIxAJhLVXHBdMPQP3dcHme-lasuweA
```
Comment (malabya) [inline config/core.entity_form_display.media.image.media_library.yml]: Remove the config hash

File: config/core.entity_view_display.media.document.download_document.yml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    - media.type.document
+  module:
+    - layout_builder
+    - media_entity_download
+third_party_settings:
+  layout_builder:
+    enabled: false
+    allow_custom: false
+_core:
+  default_config_hash: Ue2D9UBZwS3Brs8Z06aU9o2B8a6OLpSbd83kmX9bkfU
```
Comment (malabya) [inline config/core.entity_view_display.media.document.download_document.yml]: Remove the config hash

File: config/core.entity_view_display.media.image.default.yml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  config:
+    - field.field.media.image.field_media_caption
+    - field.field.media.image.field_media_credit
+    - field.field.media.image.field_media_image
+    - media.type.image
+  module:
+    - image
+    - text
+_core:
+  default_config_hash: 6PXJwTcScvkqQM-_K0wum6cmo84GAHXCr1rUKKC2FWw
```
Comment (malabya) [inline config/core.entity_view_display.media.image.default.yml]: Remove the config hash

File: config/core.entity_view_display.media.image.media_library.yml [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  module:
+    - image
+    - layout_builder
+    - text
+third_party_settings:
+  layout_builder:
+    enabled: false
+    allow_custom: false
+_core:
+  default_config_hash: 6PXJwTcScvkqQM-_K0wum6cmo84GAHXCr1rUKKC2FWw
```
Comment (malabya) [inline config/core.entity_view_display.media.image.media_library.yml]: Remove the config hash

File: config/core.entity_view_display.media.video.autoplay.yml [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    - media.type.video
+  module:
+    - file
+    - layout_builder
+third_party_settings:
+  layout_builder:
+    enabled: false
+    allow_custom: false
+_core:
+  default_config_hash: SxvbuGh-6cQMxl9bBV27-hGI46u7ZvwlMm5ObaJMNnw
```
Comment (malabya) [inline config/core.entity_view_display.media.video.autoplay.yml]: Remove the config hash

File: config/field.field.media.image.field_media_tags.yml [hunk 10]
```diff
@@ -0,0 +1,30 @@
+langcode: en
+status: true
+dependencies:
+  config:
+    - field.storage.media.field_media_tags
+    - media.type.image
+    - taxonomy.vocabulary.tags_media
+_core:
+  default_config_hash: Gi4snaFd5b4jAa3LkkBv9JcAEcurRLFLzKjrNePQSWE
```
Comment (malabya) [inline config/field.field.media.image.field_media_tags.yml]: Remove the config hash

File: config/field.storage.media.field_media_tags.yml [hunk 11]
```diff
@@ -0,0 +1,21 @@
+langcode: en
+status: true
+dependencies:
+  module:
+    - media
+    - taxonomy
+_core:
+  default_config_hash: 1Nb3KszP6ysYgX7RuIqjVUB6aNov1GrmpHpcqYijrcI
```
Comment (malabya) [inline config/field.storage.media.field_media_tags.yml]: Remove the config hash

File: recipe.yml [hunk 12]
```diff
@@ -1,9 +1,35 @@
...
+  - responsive_image
+  - lazy
+  - taxonomy
+  - media_file_delete
+config:
+  actions:
+    # Set permissions for anonymous role.
+    user.role.anonymous:
+      ensure_exists:
+        id: anonymous
```
Comment (malabya) [inline recipe.yml]: anonymous role will exist, we don't need a check for that

File: recipe.yml [hunk 13]
```diff
@@ -1,9 +1,35 @@
...
+      grantPermissions:
+        - 'download media'
+        - 'view media'
+    # Set permissions for authenticated role.
+    user.role.authenticated:
+      ensure_exists:
+        id: authenticated
```
Comment (malabya) [inline recipe.yml]: Same for authenticated

File: composer.json [hunk 14]
```diff
@@ -12,6 +12,8 @@
         "drupal/focal_point": "^2.1",
         "drupal/webp": "^1.0@RC",
         "drupal/media_entity_download": "^2.3",
-        "drupal/filefield_paths": "^1.0@beta"
+        "drupal/filefield_paths": "^1.0@beta",
+        "drupal/lazy": "3.x-dev",
```
Comment (malabya) [inline composer.json]: The `lazysizes` bower asset is required to install lazy?

Comment (mvpsatyam) [reply composer.json]: In the module README file, it says so, [Lazy](https://git.drupalcode.org/project/lazy), I only used composer command & installed it

Comment (malabya) [reply composer.json]: Yeah the library has to be installed via composer

**Pull request merged**